packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/api-client/iroha-api-client.ts

```diff
@@ -0,0 +1,247 @@
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.wsApiHost = options.wsApiHost || options.basePath || location.host;
+    this.wsApiPath = options.wsApiPath || Constants.SocketIoConnectionPathV1;
+    this.log.debug(`Created ${this.className} OK.`);
+    this.log.debug(`wsApiHost=${this.wsApiHost}`);
+    this.log.debug(`wsApiPath=${this.wsApiPath}`);
+    this.log.debug(`basePath=${this.options.basePath}`);
+
+    Checks.nonBlankString(
+      this.wsApiHost,
+      `${this.className}::constructor() wsApiHost`,
+    );
+    Checks.nonBlankString(
+      this.wsApiPath,
+      `${this.className}::constructor() wsApiPath`,
+    );
+  }
+
+  /**
+   * Start monitoring for new blocks on the Iroha ledger.
+   * @param monitorOptions - Options to be passed to validator `startMonitoring` procedure.
+   * @returns RxJs Observable, `next` - new block, `error` - any error from the validator.
+   */
+  public watchBlocksV1(
+    monitorOptions?: Record<string, unknown>,
+  ): Observable<IrohaBlockProgress> {
+    const socket: Socket = io(this.wsApiHost, { path: this.wsApiPath });
+    const subject = new ReplaySubject<IrohaBlockProgress>(0);
+    this.log.debug(monitorOptions);
+    socket.on(IrohaSocketSessionEvent.Next, (data: IrohaBlockProgress) => {
+      subject.next(data);
+    });
+
+    socket.on("connect", () => {
+      this.log.debug("connected OK...");
+      socket.emit(IrohaSocketSessionEvent.Subscribe, monitorOptions);
+    });
+
+    socket.connect();
+
+    socket.on("connect_error", (err: Error) => {
+      this.log.error("Error (connect_error): ", err);
+      socket.disconnect();
+      subject.error(err);
+    });
+
+    socket.on("connect_timeout", (err: Record<string, unknown>) => {
+      this.log.error("Error (connect_timeout): ", err);
+      socket.disconnect();
+      subject.error(err);
+    });
+
+    return subject.pipe(
+      finalize(() => {
+        this.log.info("FINALIZE - unsubscribing from the stream...");
+        socket.emit(IrohaSocketSessionEvent.Unsubscribe);
+        socket.disconnect();
+      }),
+      share(),
+    );
+  }
+
+  /**
+   * Immediately sends request to the validator, doesn't report any error or responses.
+   * @param args - arguments.
+   * @param method - function / method to be executed by validator.
+   * @param baseConfig - baseConfig needed to properly connect to ledger
+
+   */
+  public sendAsyncRequest(
+    method: Record<string, unknown>,
+    args: any,
+    baseConfig?: IrohaBaseConfig,
+  ): void {
+    this.log.debug(`inside: sendAsyncRequest()`);
+    this.log.debug(`baseConfig=${baseConfig}`);
+    this.log.debug(`methodName=${method.methodName}`);
+    this.log.debug(`args=${args}`);
+
+    if (baseConfig === undefined || baseConfig === {}) {
+      throw new RuntimeError("baseConfig object must exist and not be empty");
+    }
+
+    if (
```

Comment:
@outSH Is it okay if these fields are `null` instead of `undefined`? These current checks are not protecting against that.

Reply:
@petermetz It seems they are marked as non nullable in openapi, so not OK

Reply:
@outSH Thanks for the fix!